### PR TITLE
feat: support pre initialize number

### PR DESCRIPTION
**Pull request by RobinCodeX**
TS will automatically assign number values to members of the enum that are not assigned a value when it is compiled.
So I have many unassigned enum members in my d.ts files, and I think adding this feature.

### CODE REVIEW

File: src/typescriptUtils.ts [hunk 1]
```diff
@@ -31,7 +31,45 @@ export const getConstEnumsFromTsConfig = (tsconfigPath: string) => {
...
+        }
+      }
+      if (init) {
+        node = ts.factory.updateEnumDeclaration(
+          node,
+          node.modifiers,
+          node.name,
+          newMembers
+        );
+      }
```
Comment (rikkit) [inline src/typescriptUtils.ts:72]: I reckon this code needs an explainer comment on what its purpose is. Can you add one please?

**Review by rikkit — CHANGES_REQUESTED** [review]
Hey @RobinCodeX , thanks for the PR.

First off, I no longer need to use this package, so I don't have a good way to test regressions.

Can I just check I understand the use case correctly - if an enum member is not initialised with a numeric value, then this code will assign a value according to the same (?) rules that TypeScript assigns at compile-time. 

If that's correct, that makes sense and I'm happy to merge and release a new package if you can add an explainer comment to the new code. Cheers

File: src/typescriptUtils.ts (hunk 1 above)
Comment (RobinCodeX) [reply src/typescriptUtils.ts]: I've changed it and add debug option, I have a lot of enum, and the output of console.debug is too much.

Comment (RobinCodeX) [reply src/typescriptUtils.ts]: I have adjusted the test example and fixed some errors.

File: sample/numericSwitch/types.d.ts [hunk 2]
```diff
@@ -5,4 +5,11 @@ declare const enum HelloEnum {
   Hej = 3,
   Hola = 4,
   Ciao = 5,
-}
+  A,
+  B,
+  C,
+  D = 10,
+  EX = "EX",
+  E = 11,
```
Comment (rikkit) [inline sample/numericSwitch/types.d.ts:13]: Just wondering if this is a case that's not handled, what is the expected value of `MyEnum.C` in the following enum?

```ts
enum MyEnum {
  A = 2,
  B = "STRING VAL",
  C,
}
```

Comment (RobinCodeX) [reply sample/numericSwitch/types.d.ts:13]: This throw an error from [typescriptUtils.ts#L58](https://github.com/RobinCodeX/babel-plugin-declare-const-enum/blob/next/src/typescriptUtils.ts#L58).
I have added a new test sample [errorEnum](https://github.com/RobinCodeX/babel-plugin-declare-const-enum/tree/next/sample/errorEnum)

**Pull request closed without merging**